generator/shared.go

```diff
@@ -339,39 +339,47 @@ type GenOpts struct {
 	Copyright         string
 }
 
-// TargetPath returns the target path relative to the server package
+// TargetPath returns the target generation path relative to the server package
+// May be used in templates. Errors are not fatal: an empty string is returned instead
 func (g *GenOpts) TargetPath() string {
 	tgtAbs, err := filepath.Abs(g.Target)
 	if err != nil {
-		log.Fatalln(err)
+		log.Printf("could not evaluate target generation path \"%s\": you must create the target directory beforehand: %s", g.Target, err.Error())
+		return ""
 	}
 	srvrAbs, err := filepath.Abs(g.ServerPackage)
 	if err != nil {
-		log.Fatalln(err)
+		log.Printf("could not evaluate target server path \"%s\": %s", g.ServerPackage, err.Error())
+		return ""
 	}
 	tgtRel, err := filepath.Rel(srvrAbs, tgtAbs)
 	if err != nil {
-		log.Fatalln(err)
+		log.Printf("Target path \"%s\" and server path \"%s\" are not related. You shouldn't specify an absolute path in --server-package: %s", g.Target, g.ServerPackage, err.Error())
+		return ""
 	}
 	return tgtRel
 }
 
 // SpecPath returns the path to the spec relative to the server package
+// May be used in templates. Errors are not fatal: an empty string is returned instead
 func (g *GenOpts) SpecPath() string {
```

Comment:
why are errors not fatal, this seems like a fairly critical piece of info

Reply:
Same as TargetPath() method.